src/bp/ui-studio/src/web/views/OneFlow/diagram/ContentForm/index.tsx

```diff
@@ -0,0 +1,115 @@
+  switch (action.type) {
+    case 'fetchSuccess':
+      const { data } = action
+      return {
+        ...state,
+        contentTypes: data.map(type => ({ value: type.id, label: lang.tr(type.title) })),
+        contentTypesFields: data.reduce((acc, type) => ({ ...acc, [type.id]: type.schema.newJson }), {})
+      }
+    default:
+      throw new Error(`That action type isn't supported.`)
+  }
+}
+
+const ContentForm: FC<Props> = ({ editingContent, close, formData, onUpdate, deleteContent }) => {
+  const [state, dispatch] = useReducer(fetchReducer, {
+    contentTypes: [],
+    contentTypesFields: {}
+  })
+  const contentType = useRef(formData?.contentType || 'builtin_text')
+  const [showOptions, setShowOptions] = useState(false)
+  const [forceUpdate, setForceUpdate] = useState(false)
+  const { contentTypes, contentTypesFields } = state
+
+  useEffect(() => {
+    axios
+      .get(`${window.BOT_API_PATH}/content/types`)
+      .then(({ data }) => {
+        dispatch({
+          type: 'fetchSuccess',
+          data: data.filter(type => type.schema.newJson?.displayedIn.includes('sayNode'))
+        })
+      })
+      .catch(() => {})
+  }, [])
+
+  useEffect(() => {
+    contentType.current = formData?.contentType || 'builtin_text'
+    setForceUpdate(!forceUpdate)
+  }, [editingContent])
+
+  const moreOptionsItems: MoreOptionsItems[] = [
+    {
+      icon: 'trash',
+      label: lang.tr('module.qna.contentForm.deleteContent'),
+      action: deleteContent,
+      type: 'delete'
+    }
+  ]
+
+  const handleContentTypeChange = value => {
+    contentType.current = value
+    setForceUpdate(!forceUpdate)
+    onUpdate({ ...ContentForms.getEmptyFormData(value), contentType: value, id: formData?.id })
+  }
+
+  const contentFields = contentTypesFields?.[contentType.current]
+
+  return (
+    <RightSidebar className={style.wrapper} canOutsideClickClose close={() => close(editingContent)}>
+      <Fragment key={`${contentType.current}-${editingContent}`}>
+        <div className={style.formHeader}>
+          <Tabs id="contentFormTabs">
+            <Tab id="content" title="Say" />
+          </Tabs>
+          <MoreOptions show={showOptions} onToggle={setShowOptions} items={moreOptionsItems} />
+        </div>
+        <div className={cx(style.fieldWrapper, style.contentTypeField)}>
+          <span className={style.formLabel}>{lang.tr('studio.content.contentType')}</span>
+          {!!contentTypes.length && (
+            <Dropdown
+              filterable={false}
+              className={style.formSelect}
+              items={contentTypes}
+              defaultItem={contentType.current}
+              rightIcon="chevron-down"
+              onChange={option => {
+                handleContentTypeChange(option.value)
+              }}
+            />
+          )}
+        </div>
+        {!!contentFields && (
+          <ContentForms.Form
+            bp={{ axios, mediaPath: `${window.BOT_API_PATH}/media` }}
```

Comment:
is this suppose to handle custom path too ?

Reply:
yeah, in Q&A it uses the path set to axios since it's a module, but here axios does not have that config so it needs to get the path sent